.github/workflows/update-tech-stack.yml

```
@@ -3,25 +3,30 @@ name: Update Tech Stack
     - cron: '0 0 * * 0'  # Run weekly on Sunday at midnight
+  workflow_dispatch:  # Allow manual trigger
 
 jobs:
   update-tech-stack:
     runs-on: ubuntu-latest
+    permissions:
+      contents: write  # Explicitly grant write permission
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3  # Updated to v3
```

actionlint (1.7.4)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)